--- DirectFB-1.6.3/src/core/surface.c.zlib	2013-01-31 15:45:05.673318987 +0100
+++ DirectFB-1.6.3/src/core/surface.c	2013-01-31 15:45:38.773915790 +0100
@@ -55,6 +55,9 @@
 #include <gfx/convert.h>
 #include <gfx/util.h>
 
+#ifdef USE_ZLIB
+#include <zlib.h>
+#endif
 
 D_DEBUG_DOMAIN( Core_Surface,         "Core/Surface",         "DirectFB Core Surface" );
 D_DEBUG_DOMAIN( Core_Surface_Updates, "Core/Surface/Updates", "DirectFB Core Surface Updates" );
