--- interfaces/IDirectFBImageProvider/Makefile.am.svg	2012-07-28 09:32:26.000000000 +0000
+++ interfaces/IDirectFBImageProvider/Makefile.am	2012-07-28 09:32:48.000000000 +0000
@@ -67,7 +67,7 @@
 	-I$(top_srcdir)/lib	\
 	-I$(top_srcdir)/src
 
-AM_CPPFLAGS = -DDATADIR=\"${RUNTIME_SYSROOT}@DATADIR@\" $(LIBJPEG_CFLAGS) $(LIBPNG_CFLAGS)
+AM_CPPFLAGS = -DDATADIR=\"${RUNTIME_SYSROOT}@DATADIR@\" $(LIBJPEG_CFLAGS) $(LIBPNG_CFLAGS) $(SVG_CFLAGS)
 
 idirectfbimageprovider_LTLIBRARIES = \
 	libidirectfbimageprovider_dfiff.la	\
